lib/util.js

```diff
@@ -501,16 +501,22 @@ function formatObject(ctx, value, recurseTimes, visibleKeys, keys) {
 
 function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
-  for (var i = 0, l = value.length; i < l; ++i) {
-    if (hasOwnProperty(value, String(i))) {
-      output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
-          String(i), true));
-    } else {
-      output.push('');
+  var isSparse = keys.length * 10 < value.length;
```

Comment:
This has to be explained in the comments. How this calculation determines if the array is sparse or not?